presto-spi/src/main/java/com/facebook/presto/spi/PageBuilder.java

```diff
@@ -43,17 +43,27 @@ public PageBuilder(List<? extends Type> types)
 
     public PageBuilder(int initialExpectedEntries, List<? extends Type> types)
     {
+        this(initialExpectedEntries, DEFAULT_MAX_PAGE_SIZE_IN_BYTES, types);
+    }
+
+    public static PageBuilder withMaxPageSize(int maxPageBytes, List<? extends Type> types)
+    {
+        return new PageBuilder(Integer.MAX_VALUE, maxPageBytes, types);
+    }
+
+    private PageBuilder(int initialExpectedEntries, int maxPageBytes, List<? extends Type> types)
```

Comment:
But we don't use INT_MAX for `initialExpectedEntries` in the past. When I was trying to read through to understand to implications of that, I find the implementation not reasonable.